syntax="proto3";

package System;

service SDDFactory {
  rpc GetTask(GetTaskRequest) returns (GetTaskResponse) {}
  rpc PutResult(PutResultRequest) returns (PutResultResponse) {}
}



enum ResponseStatus {
  SUCCESS = 0;
  TRY_LATER = 1;
  TOOL_FAILED = 2;
  UNKNOWN_PROJECT = 3;
  INTERNAL_ERROR = 4;
}



enum TaskType {
  NONE = 0;
  ENERGY = 1;
  RESP = 2;
  ENERGY_RESP = 3;
}

message Task {
  TaskType task_type = 1;
  string task_identifier = 2;
  string task_content = 3; 
}

message TaskResult {
  TaskType task_type = 1;     
  string task_identifier = 2; 
  string task_result = 3;
}



message GetTaskRequest {
  string project_id = 1;
  string client_identifier = 2;
}

message GetTaskResponse {
  ResponseStatus status = 1;
  Task task = 2;
}

message PutResultRequest {
  ResponseStatus status = 1;
  TaskResult task_result = 2;
}

message PutResultResponse {
  ResponseStatus status = 1;
}
